torchdata/nodes/samplers/multi_node_weighted_sampler.py

```diff
@@ -126,11 +133,16 @@ def reset(self, initial_state: Optional[Dict[str, Any]] = None):
                 seed=self.seed,
                 rank=self.rank,
                 world_size=self.world_size,
+                epoch=self.epoch,
             )
             self._datasets_exhausted = {key: False for key in self.weights.keys()}
             for k in self.dataset_names:
                 self.source_nodes[k].reset()
 
+    def set_epoch(self, epoch: int) -> None:
+        self.epoch = epoch
+        self.reset(self._initial_state)
+
```

Comment:
Let's discuss this a bit more, i don't think we should call reset implicitly here as it will recursively reset all sub nodes